src/steamship/agents/llms/replicate.py

```diff
@@ -0,0 +1,150 @@
+
+    See the Replicate plugin source code for more information:
+      https://github.com/steamship-plugins/replicate-llms
+    """
+
+    generator: PluginInstance
+    client: Steamship
+
+    def __init__(
+        self,
+        client,
+        model_name: str = None,  # The default in the plugin is LLAMA_2_70B.
+        temperature: float = None,
+        model_version: Optional[str] = None,
+        default_system_prompt: Optional[str] = None,
+        max_tokens: Optional[int] = None,
+        top_p: Optional[float] = None,
+        stop_sequences: Optional[str] = None,  # Should be a CSV string
+        *args,
+        **kwargs
+    ):
+        """Create a new instance.
+
+        Valid model names are presently:
+         - replicate/llama-2-70b-chat
+         - replicate/dolly-v2-12b
+        """
+
+        replicate_config = remove_none(
+            {
+                "model_name": model_name,
+                "model_version": model_version,
+                "default_system_prompt": default_system_prompt,
+                "max_tokens": max_tokens,
+                "temperature": temperature,
+                "top_p": top_p,
+                "stop_sequences": stop_sequences,
+            }
+        )
+
+        generator = client.use_plugin(PLUGIN_HANDLE, config=replicate_config)
+        super().__init__(client=client, generator=generator, **kwargs)
+
+    def complete(self, prompt: str, **kwargs) -> List[Block]:
+        """Completes the prompt."""
+        options = remove_none(
+            {
+                "model_name": kwargs.get("model_name", None),
+                "model_version": kwargs.get("model_version", None),
+                "default_system_prompt": kwargs.get("default_system_prompt", None),
+                "max_tokens": kwargs.get("max_tokens", None),
+                "temperature": kwargs.get("temperature", None),
+                "top_p": kwargs.get("top_p", None),
+                "stop_sequences": kwargs.get("stop_sequences", None),
+            }
+        )
+        action_task = self.generator.generate(text=prompt, options=options)
+        action_task.wait()
+        return action_task.output.blocks
+
+
+class ChatReplicate(ChatLLM, Replicate):
+    """ChatLLM that uses Steamship's Replicate plugin to generate chat completions."""
+
+    def __init__(
+        self,
+        client,
+        model_name: str = None,  # The default in the plugin is LLAMA_2_70B.
+        temperature: float = None,
+        model_version: Optional[str] = None,
+        default_system_prompt: Optional[str] = None,
+        max_tokens: Optional[int] = None,
+        top_p: Optional[float] = None,
+        stop_sequences: Optional[str] = None,  # Should be a CSV string
+        *args,
+        **kwargs
+    ):
+        """Create a new instance.
+
+        Valid model names are presently:
+         - replicate/llama-2-70b-chat
+         - replicate/dolly-v2-12b
+        """
+
+        replicate_config = remove_none(
```

Comment:
I don't think you need to do the interim config here; passing the params along to `super().__init__` should be enough since it also removes none there.